### ISSUE TITLE
False positive not-an-iterable for apt_pkg module

### ISSUE THREAD

**Issue by bdrung** [PyCQA/pylint#7122]
### Bug description

Pylint incorrectly complains about `not-an-iterable` for Apport using the `apt_pkg` module. Following `example.py` is a distilled version of what Apport does:

```python
#!/usr/bin/python3
# pylint: disable=missing-docstring

import os

import apt

src_records = apt.apt_pkg.SourceRecords()
assert not hasattr(src_records, "binaries")
src_records.lookup("bash")
assert hasattr(src_records, "binaries")
print(f"type(src_records.binaries) = {type(src_records.binaries)}")
print(f"src_records.binaries = {src_records.binaries!r}")
pkgs = [p for p in src_records.binaries if not p.endswith("-doc")]
print(f"pkgs = {pkgs!r}")

fetcher = apt.apt_pkg.Acquire(apt.progress.text.AcquireProgress())
cache = apt.Cache(rootdir=os.getcwd())
cache.fetch_archives(fetcher=fetcher)
for i in fetcher.items:
    print(i)
```

Output of the script when executed:
```
type(src_records.binaries) = <class 'list'>
src_records.binaries = ['bash', 'bash-static', 'bash-builtins', 'bash-doc']
pkgs = ['bash', 'bash-static', 'bash-builtins']
Fetched 0 B in 0s (0 B/s)
```

### Configuration

```ini
[MASTER]
extension-pkg-whitelist=apt_pkg
```

### Command used

```shell
pylint --extension-pkg-whitelist=apt_pkg example.py
```

### Pylint output

```shell
************* Module example
example.py:14:19: E1133: Non-iterable value src_records.binaries is used in an iterating context (not-an-iterable)
example.py:20:9: E1133: Non-iterable value fetcher.items is used in an iterating context (not-an-iterable)
```

### Expected behavior

Pylint should show no complains.

### Pylint version

```shell
pylint 2.14.4
astroid 2.11.6
Python 3.10.5 (main, Jun  8 2022, 09:26:22) [GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04 (jammy)

**Comment by jacobtylerwalls** [comment]
Could you add instructions to the dependencies part of the bug report to include the package necessary to reproduce? `apt_pkg` is not a project on PyPi.

**Comment by bdrung** [comment]
Updated description. You need to have a Debian/Ubuntu based system. Then you can install the packages:
```
sudo apt install -y python3 python3-apt
```

**Comment by Pierre-Sassoulas** [comment]
There must be another step, because it seems I have apt but I still get ``a.py:6:0: E0401: Unable to import 'apt' (import-error)
`` (python error:``ModuleNotFoundError: No module named 'apt'``).

``sudo apt install -y python3 python3-apt``
```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
python3 set to manually installed.
python3-apt is already the newest version (2.0.0ubuntu0.20.04.7).
python3-apt set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

``uname -srv ``
```
Linux 5.13.0-52-generic #59~20.04.1-Ubuntu SMP Thu Jun 16 21:21:28 UTC 2022
```

**Comment by bdrung** [comment]
Based on the package versions, I assume that you are running Ubuntu 20.04 (focal). I tested my steps in a minimal Ubuntu 20.04 (focal) chroot. The result are identical to my bug description. Installed packages:
```
(focal)root@deep-thought:~# dpkg -l | grep '\(python3-apt\|python3\)  '
ii  python3                    3.8.2-0ubuntu2               amd64        interactive high-level object-oriented language (default python3 version)
ii  python3-apt                2.0.0ubuntu0.20.04.7         amd64        Python 3 interface to libapt-pkg
```

You can check (with `dpkg -L`) the content of the installed package:
```
(focal)root@deep-thought:~# dpkg -L python3-apt | grep apt/__init__.py
/usr/lib/python3/dist-packages/apt/__init__.py
```

I have no idea why you get a `ModuleNotFoundError`.
